src/utils/discord.ts

```diff
@@ -0,0 +1,104 @@
+import discord from 'discord.js';
+import { TournamentId } from '.prisma/client';
+import { fetchTournaments, fetchTournament } from '../operations/tournament';
+import { fetchTeams } from '../operations/team';
+import env from './env';
+
+// Create bot as a discord client
+const bot = new discord.Client();
+
+const createChannel = async (channelName: string, channelType: 'voice' | 'text', tournamentId: TournamentId) => {
+  // Get channel and create channels in it
+  const server = bot.guilds.cache.get(env.discord.server);
+  const channel = await server.channels.create(channelName, {
+    type: channelType,
+  });
+
+  const tournament = await fetchTournament(tournamentId);
+  await channel.setParent(tournament.discordCategoryId);
+
+  // Configure channel's permissions
+  await channel.createOverwrite(channel.guild.roles.everyone, {
+    VIEW_CHANNEL: false,
+  });
+
+  await channel.createOverwrite(
+    channel.guild.roles.cache.find((r) => r.name === channelName),
+    {
+      VIEW_CHANNEL: true,
+    },
+  );
+
+  await channel.createOverwrite(
+    channel.guild.roles.cache.find((r) => r.id === tournament.discordRespRoleId),
+    {
+      VIEW_CHANNEL: true,
+    },
+  );
```

Comment:
```suggestion
  await channel.setParent(tournament.discordCategoryId);

  // Configure channel's permissions
  const denyEveryonePromise = channel.createOverwrite(channel.guild.roles.everyone, {
    VIEW_CHANNEL: false,
  });

  const allowTeamPromise = channel.createOverwrite(
    channel.guild.roles.cache.find((r) => r.name === channelName),
    {
      VIEW_CHANNEL: true,
    },
  );

  const allowRespPromise = channel.createOverwrite(
    channel.guild.roles.cache.find((r) => r.id === tournament.discordRespRoleId),
    {
      VIEW_CHANNEL: true,
    },
  );
  
  return Promise.all([denyEveryonePromise, allowTeamPromise, allowRespPromise])
```

Si je ne me trompe pas, ces 4 actions peuvent être effectuées en même temps et ne sont pas dépendantes les unes des autres. Tu peux donc passer par un `Promise.all()`

Reply:
pour ce qui est des caractères spéciaux et des espaces, discord le gère assez bien, tu peux même mettre des emojis dans les rôles, dans les noms de channel donc si tu as gardé le fonction pour virer un bon nombre de caractères spéciaux, il ne devrait pas y avoir de problèmes à ce niveau là.

Reply:
ok parfait ! Juste pour info, tu as par contre répondu au mauvais endroit. J'avais fait ce commentaire en général, donc il vaut mieux que tu répondes en général 😊